client/lib/deploy.rs

```diff
@@ -76,31 +76,37 @@ impl From<GetBlockResult> for ListDeploysResult {
 pub(super) enum OutputKind<'a> {
     File {
         path: &'a str,
+        tmp_path: String,
         overwrite_if_exists: bool,
     },
     Stdout,
 }
 
 impl<'a> OutputKind<'a> {
     pub(super) fn file(path: &'a str, overwrite_if_exists: bool) -> Self {
+        let tmp_path = [path.clone(), ".tmp"].join("");
         OutputKind::File {
             path,
+            tmp_path,
             overwrite_if_exists,
         }
     }
 
-    pub(super) fn get(self) -> Result<Box<dyn Write>> {
+    pub(super) fn get(&self) -> Result<Box<dyn Write>> {
         match self {
             OutputKind::File {
                 path,
+                tmp_path,
                 overwrite_if_exists,
+                ..
             } => {
                 let path = PathBuf::from(path);
+                let tmp_path = PathBuf::from(tmp_path);
```

Comment:
We won't need this if `tmp_path` is changed to be a `PathBuf`, but if this were still needed, I'd suggest moving it to just before it's actually used.  I think it's better to not create variables until they're actually needed.